core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/PartialSuccessHandler.java

```diff
@@ -114,27 +114,36 @@ boolean validateTransmissionAndSend(TransmissionHandlerArgs args) {
 
 		if (args.getTransmission().getWebContentEncodingType() == "gzip") {
 
+			GZIPInputStream gis = null;
+			BufferedReader bufferedReader = null;
+
 			try {
-				GZIPInputStream gis = new GZIPInputStream(
+				gis = new GZIPInputStream(
 						new ByteArrayInputStream(args.getTransmission().getContent()));
-				BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(gis));
+				bufferedReader = new BufferedReader(new InputStreamReader(gis));
 				String line;
 				while ((line = bufferedReader.readLine()) != null) {
 					originalItems.add(line);
 				}
+			} catch (IOException ex) {
+				InternalLogger.INSTANCE.error("IOException: Error while reading the GZIP stream.%nStack Trace:%n%s", ExceptionUtils.getStackTrace(ex));
+			} catch (Throwable t) {
+				InternalLogger.INSTANCE.error("Error while reading the GZIP stream.%nStack Trace:%n%s",	ExceptionUtils.getStackTrace(t));
+			} finally {
 				if (gis != null) {
```

Comment:
If you want, you can inline `gis` and remove this block. `InputStreamReader` and `BufferedReader` both close thier respective reader/stream in each of their `close` methods.

Reply:
@littleaj yes I would like to see this change. Thanks for pointing out. It's not blocking but these are good performance constructs!

Reply:
@littleaj - thanks, will follow up with you in person